WalletWasabi/WabiSabi/Backend/Rounds/Utils/CoinJoinTransactionArchiver.cs

```diff
@@ -0,0 +1,66 @@
+using NBitcoin;
+using System;
+using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace WalletWasabi.WabiSabi.Backend.Rounds.Utils
+{
+	public class CoinJoinTransactionArchiver
+	{
+		private static readonly JsonSerializerOptions Options = new()
+		{
+			WriteIndented = true, // Pretty print.
+		};
+
+		public CoinJoinTransactionArchiver(string directoryPath)
+		{
+			BaseStoragePath = directoryPath;
+		}
+
+		/// <summary>Base path where to store transactions.</summary>
+		private string BaseStoragePath { get; }
+
+		/// <summary>
+		/// Save CoinJoin transaction in a JSON file for later inspection.
+		/// </summary>
+		/// <param name="transaction">Transaction to store.</param>
+		/// <param name="currentDate">Date when the transaction was created or <c>null</c> to use current UTC time.</param>
+		/// <returns>Full path to the stored file.</returns>
+		public async Task<string> StoreJsonAsync(Transaction transaction, DateTimeOffset? currentDate = null)
+		{
+			currentDate ??= DateTimeOffset.UtcNow;
+
+			long unixTimeStampMs = currentDate.Value.ToUnixTimeMilliseconds();
+			string txBytes = transaction.ToHex();
+
+			// Serialize entry.
+			TransactionInfo entry = new(unixTimeStampMs, transaction.GetHash().ToString(), txBytes);
+			string json = JsonSerializer.Serialize(entry, Options);
+
+			// Use a date format in the file name to let the files be sorted by date by default.
+			string fileName = $"CoinJoinTransaction.{currentDate.Value:yyyy.MM.dd-HH-mm-ss}.{transaction.GetHash()}.json";
```

Comment:
The `fileName` feels too much for me.

- `CoinJoinTransaction.` is too long prefix, It takes away so much space from the important infos. Maybe `CJTx+date` would be enough.
- Having the tx hash in the file name feels redundant. It's really annoying to copy it from the filename + it's in the json file itself.

Reply:
I have made it shorter.